### PR TITLE
remove fs and path for shader loading

### PR DESCRIPTION
**Pull request by gnavvy**
Per discussion, instead of using readFileSync(join(__dirname, ‘*.glsl'), 'utf8’) for shader source loading, change .glsl files to .glsl.js files that exports a static shader source string.

Pros:
- no more fs & path dependency for shader loading.
- no need to config webpack for brfs-babel and transform-loader if we were to import and debug layers from ‘../../src’.
- we can simply import the shader files.

Cons:
- no syntax highlighting for the .glsl file by default, we need comment out the first line for syntax highlighting

### CODE REVIEW

**Comment by gnavvy** [comment]
ran into an issue due to mapbox 0.32.1 + webpack, now testing against 0.33, will fix in the next diff.

File: src/layers/core/solid-polygon-layer/solid-polygon-layer.js [hunk 1]
```diff
@@ -60,13 +60,10 @@ const defaultProps = {
...
   getShaders() {
-    const vs64 = readFileSync(join(__dirname, './solid-polygon-layer-vertex-64.glsl'), 'utf8');
-    const vs32 = readFileSync(join(__dirname, './solid-polygon-layer-vertex.glsl'), 'utf8');
-    const fs = readFileSync(join(__dirname, './solid-polygon-layer-fragment.glsl'), 'utf8');
     return enable64bitSupport(this.props) ? {
-      vs: vs64, fs, modules: ['fp64', 'project64', 'lighting']
+      vs: solidPolygon64Vertex, solidPolygonFragment, modules: ['fp64', 'project64', 'lighting']
     } : {
-      vs: vs32, fs, modules: ['lighting']
+      vs: solidPolygonVertex, solidPolygonFragment, modules: ['lighting']
```
Comment (shaojingli) [inline src/layers/core/solid-polygon-layer/solid-polygon-layer.js]: missing "fs:"

File: src/layers/core/arc-layer/arc-layer.js [hunk 2]
```diff
@@ -40,14 +42,10 @@ const defaultProps = {
 
 export default class ArcLayer extends Layer {
   getShaders() {
-    const vs64 = readFileSync(join(__dirname, './arc-layer-64-vertex.glsl'), 'utf8');
-    const vs32 = readFileSync(join(__dirname, './arc-layer-vertex.glsl'), 'utf8');
-    const fs = readFileSync(join(__dirname, './arc-layer-fragment.glsl'), 'utf8');
-
     return enable64bitSupport(this.props) ? {
-      vs: vs64, fs, modules: ['fp64', 'project64']
+      vs: arc64Vertex, arcFragment, modules: ['fp64', 'project64']
```
Comment (shaojingli) [inline src/layers/core/arc-layer/arc-layer.js]: ditto

File: src/layers/core/grid-cell-layer/grid-cell-layer.js [hunk 3]
```diff
@@ -68,14 +70,10 @@ export default class GridCellLayer extends Layer {
    */
 
   getShaders() {
-    const vs64 = readFileSync(join(__dirname, './grid-cell-layer-64-vertex.glsl'), 'utf8');
-    const vs32 = readFileSync(join(__dirname, './grid-cell-layer-vertex.glsl'), 'utf8');
-    const fs = readFileSync(join(__dirname, './grid-cell-layer-fragment.glsl'), 'utf8');
-
     return enable64bitSupport(this.props) ? {
-      vs: vs64, fs, modules: ['fp64', 'project64', 'lighting']
+      vs: gridCell64Vertex, gridCellFragment, modules: ['fp64', 'project64', 'lighting']
```
Comment (shaojingli) [inline src/layers/core/grid-cell-layer/grid-cell-layer.js]: ditto

File: src/layers/core/scatterplot-layer/scatterplot-layer.js [hunk 4]
```diff
@@ -44,14 +46,10 @@ const defaultProps = {
 
 export default class ScatterplotLayer extends Layer {
   getShaders(id) {
-    const vs64 = readFileSync(join(__dirname, './scatterplot-layer-64-vertex.glsl'), 'utf8');
-    const vs32 = readFileSync(join(__dirname, './scatterplot-layer-vertex.glsl'), 'utf8');
-    const fs = readFileSync(join(__dirname, './scatterplot-layer-fragment.glsl'), 'utf8');
-
     return enable64bitSupport(this.props) ? {
-      vs: vs64, fs, modules: ['fp64', 'project64']
+      vs: scatterplot64Vertex, scatterplotFragment, modules: ['fp64', 'project64']
```
Comment (shaojingli) [inline src/layers/core/scatterplot-layer/scatterplot-layer.js]: missing 'fs:'

File: src/layers/core/arc-layer/arc-layer-64-vertex.glsl.js [hunk 5]
```diff
@@ -18,6 +18,8 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
+/* eslint-disable max-len */
```
Comment (ibgreen) [inline src/layers/core/arc-layer/arc-layer-64-vertex.glsl.js]: Nit: I think we have a max-len set to 100 characters in our eslint config. I'd prefer that we format our glsl code to fit within this limit. I suspect that a few carefully placed newlines would do the trick

File: src/layers/core/grid-cell-layer/grid-cell-layer-64-vertex.glsl.js [hunk 6]
```diff
@@ -1,6 +1,28 @@
...
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
 // Inspired by screen-grid-layer vertex shader in deck.gl
 
 /* vertex shader for the grid-layer */
+/* eslint-disable max-len */
```
Comment (ibgreen) [inline src/layers/core/grid-cell-layer/grid-cell-layer-64-vertex.glsl.js]: Can we remove eslint-...?

File: src/layers/core/hexagon-cell-layer/hexagon-cell-layer-fragment.glsl.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// See: npm glsl-shader-name
```
Comment (ibgreen) [inline src/layers/core/hexagon-cell-layer/hexagon-cell-layer-fragment.glsl.js]: If we are using SHADER_NAME everywhere, I don't think we need this comment in the shaders.

**Comment by gnavvy** [comment]
I forgot those in the examples, will open another PR later today.

**Comment by ibgreen** [comment]
> I forgot those in the examples, will open another PR later today.

Good catch. Also check the `shaderlib` folder.

The ultimate goal is to remove the `static-fs`, `brfs-babel` and `transform-loader` modules from all `package.json`s (both in deps and scripts) and also from webpack configs. If the code still builds and runs, that means the "purge" has been successful.

Modules that are involved in shader inlining:
```
"babel-plugin-static-fs": "^1.1.0",
"brfs-babel": "^1.0.0",
"transform-loader": "^0.2.3",
```

**Pull request merged**